frodo.js

```diff
@@ -1,24 +1,23 @@
 /**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.